### ISSUE TITLE
Add Source Address Translation support

### ISSUE THREAD

**Issue by Jeff-Gibbs** [joel74/POSH-LTM-Rest#96]
The module has been working great, until a new requirement to set Source Address Translation to Auto Map
both New-VirtualServer and Set-VirtualServer don't look like they support changing SNAT.

**Comment by joel74** [comment]
Thanks, Jeff. I'll check that out and see if we can get is resolved shortly.

**Comment by Jeff-Gibbs** [comment]
Great!!Thank You Joel

**Comment by vercellone** [comment]
Set-VirtualServer is written to support changing property values, even if they aren't exposed as a parameter.

```Powershell
$vs = Get-VirtualServer -F5Session $YourSession -Name 'your-virtualserver-name' -Partition YourPartition
$vs.sourceAddressTranslation = 'automap'
$vs | Set-VirtualServer -F5Session $YourSession
```
Let me know if that doesn't work.

**Comment by Jeff-Gibbs** [comment]
vercellone,
Nope.. Error:Invoke-F5RestMethod : "400 Bad Request: Found unexpected json string at property group /ltm/virtual/~Common~vs_test4/source-address-translation. The json string is 
"automap".

**Comment by joel74** [comment]
I'm going to add sourceAddressTranslation as a names param. I'm not a big fan of unknown / unexposed params. I also want to support setting it to SNAT and specifying a pool, as well as setting the virt server to enabled or disabled.

**Comment by joel74** [comment]
I've submitted PR#98 which updates New-VirtualServer to include source address translation settings as well as enabled/disabled. 
Jason, I'm thinking that the reason your example didn't work for Jeff-Gibbs is because it would need to be: 
$vs.sourceAddressTranslation.type = 'automap'
or something of that nature. I don't necessarily want to have to explicitly name all the possible params for virtual server objects and others, but I also want to make it as easy as possible for people to discover what settings / params are allowed.

**Comment by Jeff-Gibbs** [comment]
$vs.sourceAddressTranslation.type = 'automap'  does work with Set-VirtualServer, ultimately setting it on creation would be best.  Thanks Joel and Jason

**Comment by Jeff-Gibbs** [comment]
Joel,
I've tested changes in pull request #98 just for sourceAddressTranslation and adding automap and it works!
Thanks

**Comment by joel74** [comment]
Excellent. Thanks for testing, Jeff. I'll most likely merge this PR for the New-VirtualServer changes, but leave this open until it's decided what we want to do with Set-VirtualServer. Jason did nearly all the work on that function

**Comment by vercellone** [comment]
Don't let me stand in your way.  I am not opposed to exposing _common_ properties as parameters.  I am just opposed to exposing _all possible_ properties as parameters.  The more parameters you expose, the greater the maintenance overhead, especially when property assignment details vary between F5 versions.

**Comment by joel74** [comment]
This will be resolved with the changes made in PR #112

**Comment by joel74** [comment]
#112 has been merged.
